packages/beacon-node/src/chain/opPools/attestationPool.ts

```diff
@@ -25,7 +25,22 @@ const MAX_ATTESTATIONS_PER_SLOT = 16_384;
 type AggregateFast = {
   data: phase0.Attestation["data"];
   aggregationBits: BitArray;
-  signature: Signature;
+  /**
+   * Two potential strategies to pre-aggregate attestation signatures:
+   * - Aggregate new signature into existing aggregate on .add(). More memory efficient as only 1 signature
+   *   is kept per aggregate. However, the eager aggregation may be useless if the connected validator ends up
+   *   not being an aggregator. bls.Signature.fromBytes() is not free, thus the aggregation may be done around
+   *   the 1/3 of the slot which is a very busy period.
+   * - Defer aggregation until getAggregate(). Consumes more memory but prevents extra work by only doing the
+   *   aggregation if the connected validator is an aggregator. The aggregation is done during 2/3 of the slot
+   *   which is a less busy time than 1/3 of the slot.
+   */
+  signatures: Uint8Array[];
+  /**
+   * There could be up to TARGET_AGGREGATORS_PER_COMMITTEE aggregator per attestation data and all getAggregate() call
+   * tends to be at the same time so we want to cache the aggregated attestation, invalidate cache per new unaggregated attestation.
+   */
+  aggregatedAttestation: phase0.Attestation | null;
```

Comment:
I don't see where this is ever being set.

Reply:
oh good catch, thanks